distributed/scheduler.py

```diff
@@ -7991,15 +7999,22 @@ def check_idle(self) -> float | None:
                 self.idle_since = last_activity
                 return self.idle_since
 
-        if self.idle_timeout:
-            if time() > self.idle_since + self.idle_timeout:
+        if not self.clients:
+            self.close_for_idle_timeout(self.idle_timeout_no_clients)
+
+        self.close_for_idle_timeout(self.idle_timeout)
```

Comment:
This will print two log messages and schedule two coroutines if you exceed both idle_timeout and idle_timeout_no_clients.
Could you refactor it so that only one is fired?